core/src/main/java/org/wso2/msf4j/internal/MicroservicesServerSC.java

```diff
@@ -156,15 +160,64 @@ protected void addCarbonTransport(CarbonTransport carbonTransport) {
         Dictionary<String, String> properties = new Hashtable<>();
         properties.put(MSF4JConstants.CHANNEL_ID, carbonTransport.getId());
         microservicesRegistries.put(carbonTransport.getId(), microservicesRegistry);
-        DataHolder.getInstance().getBundleContext()
-                  .registerService(MicroservicesRegistry.class, microservicesRegistry, properties);
+        DataHolder.getInstance().getBundleContext().ifPresent(bundleContext ->
+                bundleContext.registerService(MicroservicesRegistry.class, microservicesRegistry, properties));
     }
 
     protected void removeCarbonTransport(CarbonTransport carbonTransport) {
         DataHolder.getInstance().getMicroservicesRegistries().remove(carbonTransport.getId());
     }
 
     @Reference(
+            name = "interceptor-config",
+            service = OSGiInterceptorConfig.class,
+            cardinality = ReferenceCardinality.MULTIPLE,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "removeInterceptorConfig"
+    )
+    protected void addInterceptorConfig(OSGiInterceptorConfig interceptorConfig, Map properties) {
+        Object channelId = properties.get(MSF4JConstants.CHANNEL_ID);
+        Map<String, MicroservicesRegistryImpl> microservicesRegistries =
+                DataHolder.getInstance().getMicroservicesRegistries();
+        if (channelId != null) {
+            MicroservicesRegistryImpl microServicesRegistry = microservicesRegistries.get(channelId.toString());
+            if (microServicesRegistry == null) {
+                throw new OSGiDeclarativeServiceException("Couldn't found the registry for channel ID " +
```

Comment:
It might be better to define this exception in the method declaration to let the callers know that this method may throw exceptions in certain scenarios.